### PR TITLE
Fix / Enhancement For Error Handler, Update Documentation

### LINKED ISSUE #18774: Fix / Enhancement For Error Handler in Angular
Related #18753 

when the user customize error handling in angular, the default error handler still works, keeps using default.

## Todo

- [x] Fix Customization bugs, Make easier to customize. 
- [x] Update Documentation

### PR DESCRIPTION
**Pull request by Sinan997**
Resolves #18774

### CODE REVIEW

**Review by mahmut-gundogdu — APPROVED** [review]

**Comment by masum-ulu** [comment]
## Todo
- Add functional based error handler example to http-error-handler document page
  - Make sure example works correct
- Add http-error-handler page to http-request document page as `see also` or `next`

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+- Created a service named `MyCustomErrorHandlerService`, and provided via `useExisting` key because we dont want another instance of it. And set `multi` key to true because ABP default error handlers are also provided with **CUSTOM_ERROR_HANDLERS** injection token.
+
+- 400 errors are handled from custom `MyCustomErrorHandlerService`. When a 400 error occurs, backend error message will be displayed as shown below:
+
+![custom-error-handler-toaster-message](images/custom-error-handler-toaster-message.jpg)
+
+**Note 1:** If your service cannot handle the error. Then ABP will check the next Error Service.
+**Note 2:** If none of the service handle the error. Then basic confirmation message about the error will be shown to the user.
+**Note 3:** You can provide more than one service, with CUSTOM_ERROR_HANDLER injection token.
+**Note 4:** If you want your custom service to be evaluated (checked) earlier, set the priority variable high.
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: **Do notes like below please**

### Notes
- If your service cannot handle the error. Then ABP will check the next Error Service
- If none of the service handle the error. Then basic confirmation message about the error will be shown to the user
- You can provide more than one service, with CUSTOM_ERROR_HANDLER injection token
- If you want your custom service to be evaluated (checked) earlier, set the priority variable high

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+}
+```
+
+### Service Method;
+
+With Services you can provide **more than one handler**.
+
+A custom HTTP error handler service can be registered to an injection token named **`CUSTOM_ERROR_HANDLERS`**.
+
+ABP has some default error handlers, you can see them [here](https://github.com/abpframework/abp/blob/dev/npm/ng-packs/packages/theme-shared/src/lib/providers/error-handlers.provider.ts).
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: - Remove enters 
- Remove last description `you can see theme here`, instead give ref. to the `error handlers` sentence

### Example

You can provide **more than one handler** with services. A custom HTTP error handler service can be registered with  injection token named **`CUSTOM_ERROR_HANDLERS`**. ABP has some default [error handlers](https://github.com/abpframework/abp/blob/dev/npm/ng-packs/packages/theme-shared/src/lib/providers/error-handlers.provider.ts)

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  if (httpError.status === 500) {
+    // handle 500 errors here
+  }
+
+  // you can return the of(httpError) at bottom of the function to run the default handler of ABP for HTTP errors that you didn't handle above.
+  return of(httpError);
+}
+```
+
+### Service Method;
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: Remove `;`

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
+# HTTP Error Handling
+
+When the `RestService` is used, all HTTP errors are reported to the [`HttpErrorReporterService`](./HTTP-Error-Reporter-Service), and then `ErrorHandler`, a service exposed by the `@abp/ng.theme.shared` package automatically handles the errors.
+
+## Custom HTTP Error Handler
+
+### (Deprecated!) Function Method;
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: ## Todo

- Remove `;`
- Use title like below

### Function Method `Deprecated`

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+### Service Method;
+
+With Services you can provide **more than one handler**.
+
+A custom HTTP error handler service can be registered to an injection token named **`CUSTOM_ERROR_HANDLERS`**.
+
+ABP has some default error handlers, you can see them [here](https://github.com/abpframework/abp/blob/dev/npm/ng-packs/packages/theme-shared/src/lib/providers/error-handlers.provider.ts).
+
+### How To Add New Handler Service?
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: Remove `?`

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+- ABP error handler services are implements the interface of **CustomHttpErrorHandlerService**.
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: Remove `-` item sign

**It must be loooks like below**
---

### How To Add New Handler Service
ABP error handler services are implements the interface of **CustomHttpErrorHandlerService**

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+```ts
+interface CustomHttpErrorHandlerService {
+  readonly priority: number;
+  canHandle(error: unknown): boolean;
+  execute(): void;
+}
+```
+
+- **`priority`** ABP sorts the services according to the number of the priority variable. Higher priority will be checked first. You can think this as a z-index property in CSS.
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: Remove description

`You can think this as a z-index property in CSS.`

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+- **`canHandle(error: unknown): boolean :`** Check if the service can handle the error. Returns boolean.
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: ### Remove 
- method's arg
- return type 
- `:`

### Example
- **`canHandle`** Check if the service can handle the error. Return boolean

File: docs/en/UI/Angular/HTTP-Error-Handling.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+- **`execute(): void :`** If the service can handle the error, then run the execute method.
```
Comment (masum-ulu) [inline docs/en/UI/Angular/HTTP-Error-Handling.md]: - **`execute`** If the service can handle the error, then run the execute method

**Review by masum-ulu — APPROVED** [review]

**Pull request merged**